### PR TITLE
Dependency Updates, Refactors and more code projects

### PR DESCRIPTION
**Pull request by ddyett**
This change:

- Adds some more of the code projects to the list of sdks
- Refactors a number of place
- Updates Dependencies of typescript, eslint to newer versions
- Updates the project to .NET5 and C# 9.0 (note pipelines not updated0
- Converts some javascript files to typescript

### CODE REVIEW

**Comment by ddyett** [comment]
@andrueastman @MaggieKimani1 @ElinorW  ping

File: SamplesDashboard/SamplesDashboard/Extensions/IEnumerableExtensions.cs [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+    {
+        public static string BuildQueryString(this IEnumerable<string> names)
+        {
+            var flattenedString = new StringBuilder();
+            foreach (var name in names)
+            {
+                flattenedString.Append(name + " OR ");
+            }
+
+            return flattenedString.Remove(flattenedString.Length - " OR ".Length, " OR ".Length).ToString();
```
Comment (andrueastman) [inline SamplesDashboard/SamplesDashboard/Extensions/IEnumerableExtensions.cs:16]: 👍

File: SamplesDashboard/SamplesDashboard/ClientApp/package.json [hunk 2]
```diff
@@ -35,23 +35,20 @@
...
+    "eslint-plugin-import": "2.22.1",
+    "eslint-plugin-jsx-a11y": "6.4.1",
+    "eslint-plugin-react": "7.22.0"
   },
   "scripts": {
     "start": "react-scripts start",
     "build": "react-scripts build",
     "test": "cross-env CI=true react-scripts test --env=jsdom",
     "eject": "react-scripts eject",
-    "lint": "tslint --project tsconfig.json -c tslint.json"
```
Comment (andrueastman) [inline SamplesDashboard/SamplesDashboard/ClientApp/package.json:54]: We still need this as it is used in a validation stage of the pipeline.

**Review by andrueastman — APPROVED** [review]
Thanks for cleaning this up and changing the JavaScript files to TypeScript files. 

I have fixed/updated the validation pipelines via this commit https://github.com/microsoftgraph/msgraph-samples-dashboard/commit/f630fb4707d0f38763dceb6fa5f5eb75cca625d7

Otherwise I believe we should be good to go for the .NET 5 upgrade. (We will just need to update the runtime of the relevant app service before deploying this)

**Pull request merged**